dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/CDIContainerImpl.java

```diff
@@ -582,7 +586,26 @@ public String getCurrentApplicationContextID() {
                         runtimeExtensionMap.put(serviceID, extensionArchive);
                     }
                 }
+                extensionSet.add(extensionArchive);
+            }
+        }
+
+        //Now do the exact same thing for extensions coming from the SPI
+        Iterator<ServiceAndServiceReferencePair<WebSphereCDIExtensionMetaData>> spiExtensions = cdiRuntime.getSPIExtensionServices();
```

Comment:
what happens if a class is both WebSphereCDIExtension and WebSphereCDIExtensionMetaData?

Reply:
I think there will be two BDAs. I can add a check that throws an exception for it if you'd like. (I think the odds of that happening are low since WebSphereCDIExtension isn't exposed to user features)